### DISCUSSION TITLE
0.57.0 beta 1

### DISCUSSION THREAD

**Discussion by zachowj** [zachowj/node-red-contrib-home-assistant-websocket#1077] [Beta]
**Welcome to the Home Assistant WebSocket Node-RED Beta Release Notes**

Always recommend doing a backup of your flows before upgrading.

### Breaking Changes

**Exposure As:**
We've completely overhauled the "expose as" system! Now, it uses a switch entity config, allowing a single switch entity to control multiple event nodes. Please note that this is a **breaking change**. To continue using the expose as functionality, make sure to update the node schema from the Node-RED UI.

**Trigger Service:**
We've simplified the trigger service. It now accepts only two optional data properties: `output_path` and `message`.
- `output_path`: Specify a comma-separated list of paths to send the message to. Use '0' to send the message to all paths.
- `message`: Define the message object to be sent to the next node.

**Device Node:**
In the companion integration, we've updated the endpoints. To ensure proper functionality, you'll need version 3.0.0+ for this node.

#### Bare Metal / Docker

To install or update, use npm inside the config directory:

```bash
npm install node-red-contrib-home-assistant-websocket@beta
```
or

Upload the attached file using the package manager in Node-RED. 

#### Home Assistant Add-on

Add the beta release to the [`npm_packages`](https://github.com/hassio-addons/addon-node-red/blob/main/node-red/DOCS.md#option-npm_packages) config option.

```yaml
npm_packages:
  - node-red-contrib-home-assistant-websocket@beta
```

```yaml
init_commands:
  - (cd /config/node-red/ && npm uninstall node-red-contrib-home-assistant-websocket)
```

---

The full changelog change can be found in the release pull request https://github.com/zachowj/node-red-contrib-home-assistant-websocket/pull/1070

<hr /><em>This discussion was created from the release <a href='https://github.com/zachowj/node-red-contrib-home-assistant-websocket/releases/tag/v0.57.0-beta.1'>0.57.0 beta 1</a>.</em>

**Comment by RobertD502** [comment]
Looks like adding `node-red-contrib-home-assistant-websocket@beta` to the `npm_packages` section of the Node-Red HA add-on configuration doesn't seem to install the beta version over `0.56.1`.

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/52541649/b2efa23c-961c-404c-98bb-18856dfb9bf3)

**Node-Red Palette showing version `0.56.1` being used:**

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/52541649/b2667176-5076-4ebc-b2c2-bce53726c9df)

**Node-Red addon debug:**
```
27 Sep 14:52:45 - [debug] Module: node-red-contrib-home-assistant-websocket 0.56.1 /config/node-red/node_modules/node-red-contrib-home-assistant-websocket
```


Edit: entered into the node-red addon container and can see that the `package.json` for the module does indeed show the betas as being the installed module:
```json
{
  "name": "node-red-contrib-home-assistant-websocket",
  "description": "Node-RED integration with Home Assistant through websocket and REST API",
  "version": "0.57.0-beta.1",
  "homepage": "https://zachowj.github.io/node-red-contrib-home-assistant-websocket/",
  "bugs": {
    "url": "https://github.com/zachowj/node-red-contrib-home-assistant-websocket/issues"
  },
```

For some reason, the beta nodes aren't being loaded. For example, the `Webhook` node doesn't have the `Expose as` property, confirming that the Beta isn't being loaded over 0.56.1:

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/52541649/33f518c6-e96c-4373-83f5-72d23fed53dc)

Edit 2:

## Solution
Users of the Home Assistant Node-Red `add-on`, in addition to adding the beta to the `npm_packages` config, need to edit the `package.json` and `package-lock.json` files within the `node-red` folder (located in `/config/node-red`) and make sure the `node-red-contrib-home-assistant-websocket` module has the beta version defined like this:

```
"node-red-contrib-home-assistant-websocket": "~0.57.0-beta.1"
```

After doing so, restart the Node-Red add-on.

**Reply by zachowj** [reply]
Must have changed since the last time I used the addon. Should be able to add `(cd /config/node-red/ && npm uninstall node-red-contrib-home-assistant-websocket)` to the `init_commands`.
